src/v0/destinations/tiktok_ads/util.js

```diff
@@ -60,4 +60,29 @@ const hashUserField = (user) => {
   }
   return updatedUser;
 };
-module.exports = { getContents, hashUserField };
+
+/**
+ * Get event source from message properties or channel. We have considered the following doc
+ * https://business-api.tiktok.com/portal/docs?id=1771100779668482
+ * @param {*} message
+ * @returns (string): The resolved, valid event source — one of 'web', 'app', 'offline', or 'crm'. Defaults to 'web' if none are valid.
+ */
+const getEventSource = ({ properties, channel }) => {
+  const supportedSources = ['web', 'app', 'offline', 'crm'];
+
+  const isSupported = (source) => supportedSources.includes(source);
+
+  const { eventSource } = properties || {};
```

Comment:
Is camel case right standard for attributes ?

Reply:
Shouldn't it be properties.event_source ?
Ref: https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/browsing/
We follow snake case in e-commerce spec

Reply:
For any attributes on RudderStack we generally do in camel case.

Reply:
https://www.rudderstack.com/docs/destinations/streaming-destinations/tiktok-ads/cloud-mode/#property-mapping you can refer this mapping we are prioritizing anything in camelCase.